fdbrpc/FlowTransport.actor.cpp

```diff
@@ -808,27 +858,26 @@ static void scanPackets(TransportData* transport, uint8_t*& unprocessed_begin, c
 
 	const bool checksumEnabled = !peerAddress.isTLS();
 	loop {
-		uint32_t packetLen, packetChecksum;
+		uint32_t packetLen;
 
-		//Retrieve packet length and checksum
-		if (checksumEnabled) {
-			if (e-p < sizeof(uint32_t) * 2) break;
-			packetLen = *(uint32_t*)p; p += sizeof(uint32_t);
-			packetChecksum = *(uint32_t*)p; p += sizeof(uint32_t);
-		} else {
-			if (e-p < sizeof(uint32_t)) break;
-			packetLen = *(uint32_t*)p; p += sizeof(uint32_t);
-		}
+		auto protocolChecksumIter = protocolToChecksum.upper_bound(std::min(g_network->protocolVersion(), peerProtocolVersion));
```

Comment:
It's critical to get this logic right. Would you mind adding a unit test for it? You can use the `TEST_CASE` macro from "flow/UnitTest.h"